src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.cpp

```diff
@@ -78,6 +79,12 @@ Value insertAllocAndDealloc(MemRefType type, Location loc,
 
   // Make sure to allocate at the beginning of the block if
   // all dimensions are known.
+  // Don't do these if memory bundling is not enabled
+  // Allow buffer management in MLIR to do the job
+
+  if (!memoryBundlingEnabled)
```

Comment:
I am a bit concerned about the use of an option of the `onnx-mlir` command directly inside a pass, in this case, onnx-to-krnl pass. I think that onnx-to-krnl pass should be self-contained and not depend on the `onnx-mlir` command, which is easier for debugging.

How about adding a local option, e.g. `no-deallocation`, to the `onnx-to-krnl` pass in order to enable/disable deallocation? and users of `onnx-to-krnl` can set the option if they want, e.g., 
- with `onnx-mlir-opt`: `onnx-mlir-opt --convert-onnx-to-krnl='no-deallocation'`
- with `onnx-mlir`: `mlir::createLowerToKrnlPass(/*no-deallocation=*/ true)`

Reply:
@Tungld I discussed my thinking of options in previous PR or issue about command line options for instrument. I did not find an easy way to define option with field Option in definition of pass, Similar situation to propagate options to a pass: either we have initialize the options explicitly or the default value (not the option) has to be set. With the default value set, I did not experiment how onnx-mlir-opt to use that.  Did you ever try that?
In my view, our pass may depend on some shared context info, such as target triple, and optimization/debugging level, to get consistent logic across passes. I am not too worried about using option inside a pass. On the contrary, I feel it is a better way to make onnx-mlir and onnx-mlir-opt work in the same way. The drawback is that it makes the pass in onnx mlir compiler less reusable.  I think that's the concern from MLIR, which provides basic passes. We can discuss the topic more and it may take some times for our compiler to evolve.

Reply:
I just checked the use of an option in `onnx-mlir-opt`: https://github.com/tungld/onnx-mlir/pull/1

`onnx-mlir-opt --convert-onnx-to-krnl` will insert deallocation as normal, and `onnx-mlir-opt --convert-onnx-to-krnl='no-dealloc'` will not insert deallocation.

Reply:
@tungld Thanks for trying this. Can I use the no-dealloc option for onnx-mlir or other way to turn on/off no-dealloc?
By the way, when I tried option of instrument, I found I can not use the enum OptionList for onnx-mlir-opt, while I can for onnx-mlir. Also I had difficult in giving multiple options. The document about option in pass management is not enough. Anything else?

Reply:
> Can I use the no-dealloc option for onnx-mlir or other way to turn on/off no-dealloc?

Unfortunately, we cannot use the option directly. But, we can set it via constructor. e.g. (I have not really tested this)
```
FrontendToKrnlLoweringPass(bool noDealloc) { this->noDealloc = noDealloc}
```
and then
```
std::unique_ptr<Pass> mlir::createLowerToKrnlPass(bool noDealloc) {
  return std::make_unique<FrontendToKrnlLoweringPass>(noDealloc);
}
```

The thing is we should keep the default constructor (with no arguments) to make options work. It looks like a limitation.

> When I tried option of instrument, I found I can not use the enum OptionList for onnx-mlir-opt, while I can for onnx-mlir. Also I had difficult in giving multiple options. The document about option in pass management is not enough. Anything else?
By the way, I will be on vacation next week.

I am sorry I haven't yet used OptionList. Have a nice vacation :) !